### ISSUE TITLE
[Feature]: Improved custom document scanner

### ISSUE THREAD

**Issue by astubenbord** [astubenbord/paperless-mobile#322]
### Description

While the current scanner library works, there are several limitations including customizability and bugs. Therefore, a custom scanner using platform channels and native views for camera should be implemented.

**Comment by Drexel2k** [comment]
Hey! I am using Paperless and your app for a few days, I appreciate you work very much, thank you for this. 

But I also noticed hat the scan quality is bad, I put ASN QR codes on my documents and the PDFs are so compressed that Paperless had some problems recognizing them. I noticed that Files by Google and Google Drive have a scanner option which I use now as the quality, document border recognition and post editing possibilities are pretty nice.

They offer an API for this scanner since February, it still seems beta, but the results are good: https://android-developers.googleblog.com/2024/02/ml-kit-document-scanner-api.html

Maybe this would be an option.

**Comment by cprpolo** [comment]
Congratulations on this project!! 
Loving it so far, only thing is how can i make a "tweak" on the picture/scan before upload it? In apps like CamScanner they have a magic button to brighten up the picture when needed and so on, is there a way of doing something like that in Paperless Android app?
Thanks in advance

**Comment by goncalossilva** [comment]
I miss this as well. Most scans lack contrast and look blurry in comparison with most "scanner" apps.

**Comment by ptoews** [comment]
Any update on this? A quick [search](https://github.com/search?q=+flutter+document+scanner&type=repositories&s=updated&o=desc) on Github shows a ton of flutter document scanners. Surely one of them could be used here?

**Comment by astubenbord** [comment]
@ptoews Right, there are some new projects that didn't exist back then :) But: As far as I've seen, they use google mlkit under the hood, which is prohibited by F-Droid.

**Comment by astubenbord** [comment]
Feel free to check out the alpha for the new scanner and let me know your thoughts: https://github.com/astubenbord/paperless-mobile/discussions/645

**Comment by ptoews** [comment]
@astubenbord Tried it out a few times so far, it's a huge improvement!
